GitCommands/Git/Executable.cs

```diff
@@ -98,9 +98,16 @@ public ProcessWrapper(string fileName, string arguments, string workDir, bool cr
 
                 _process.Exited += OnProcessExit;
 
-                _process.Start();
-
-                _logOperation.SetProcessId(_process.Id);
+                try
+                {
+                    _process.Start();
+                    _logOperation.SetProcessId(_process.Id);
+                }
+                catch (Exception ex)
+                {
+                    _logOperation.LogProcessEnd(ex);
```

Comment:
If an exception is thrown here the object is never initialized and thus Dispose won't happen. Disposing before rethrowing should fix that.